src/flint/test/test_all.py

```diff
@@ -2834,6 +2834,44 @@ def setbad(obj, i, val):
         if type(p) == flint.fq_default_poly:
             assert raises(lambda: p.integral(), NotImplementedError)
 
+        # resultant checks.
+
+        if is_field and characteristic == 0:
+            # Check that the resultant of two cyclotomic polynomials is right.
+            # See Dresden's 2012 "Resultants of Cyclotomic Polynomials"
+            for m in range(1, 50):
+                for n in range(m + 1, 50):
+                    a = flint.fmpz_poly.cyclotomic(m)
+                    b = flint.fmpz_poly.cyclotomic(n)
+                    q, r = divmod(flint.fmpz(n), flint.fmpz(m))
+                    fs = q.factor()
+                    if r != 0 or len(fs) > 1:
+                        assert a.resultant(b) == 1
+                    else:
+                        prime = fs[0][0]
+                        tot = flint.fmpz(m).euler_phi()
+                        assert a.resultant(b) == prime**tot
+
+        x = P([0, 1])
+
+        if composite_characteristic and type(x) == flint.fmpz_mod_poly:
+            # Flint crashes in this case, even though the resultant could be
+            # computed.
+            divisor = characteristic.factor()[0][0]
+            if type(x) == flint.fmpz_mod_poly:
+                assert raises(lambda: x.resultant(x + divisor), ValueError)
+        elif type(x) == flint.fq_default_poly:
+            # Flint does not implement resultants over GF(q) for nonprime q, so
+            # there's nothing for us to check.
+            pass
+        else:
+            assert x.resultant(x) == 0
+            assert x.resultant(x**2 + x - x) == 0
+            assert x.resultant(x**10 - x**5 + 1) == S(1)
```

Comment:
I'm surprised that nmod_poly doesn't crash here. I guess it sometimes does and sometimes doesn't:
```python
In [3]: a = flint.nmod_poly([1, 2, 3], 12)

In [4]: b = flint.nmod_poly([1, 2, 3], 12)

In [5]: a.resultant(b)
Out[5]: 0

In [6]: b = flint.nmod_poly([1, 2, 3, 4], 12)

In [7]: a.resultant(b)
Flint exception (Impossible inverse):
    Cannot invert modulo 3*4
Aborted (core dumped)
```

Reply:
Good to know! I added a similar check for nmod_poly that makes this example throw an exception.